scylla/src/policies/speculative_execution.rs

```diff
@@ -180,6 +180,11 @@ where
                     } else {
                         last_error = Some(r)
                     }
+                } else {
+                    // The only case where None is returned is when execution plan was exhausted.
+                    // If so, there is no reason to start any more fibers.
+                    // We can't always return - there may still be fibers running.
+                    retries_remaining = 0;
```

Comment:
:warning: Naming is extremely confusing. Retries are something that occurs within a single speculative fiber, whereas this should be named `spawnable_fibers_remaining` or similarly.
:wrench: I encourage you to adjust naming where possible without breaking changes. Unfortunately, `SpeculativeExecutionPolicy::{max_retry_count,retry_interval}` are set in stone until 2.0. For them, I suggest adding TODOs for 2.0 to fix their naming to avoid name clash with `RetryPolicy` abstractions.

Reply:
We can certainly discuss fully changing the naming in 2.0. For now, variable name is consistent with methods on the trait, so I don't see much point in changing it. This is internal driver code, in speculative execution module - it is imo clear that retries here refer to speculative retries, not normal ones.

Also, not related to this PR at all.

Reply:
The request is retried (speculatively, so without waiting for the result of previous attempt).

Reply:
> Will you add the TODOs I mentioned?
ping @Lorak-mmk